BLAKE2 SIMD128: bool/AltiVec keyword conflict on PowerPC with GCC + -std=c11

## Summary

`Hacl_Hash_Blake2s_Simd128.c` cannot compile on PowerPC (ppc64le) with GCC 10+ and `-std=c11 -maltivec -mvsx`. GCC's AltiVec extension redefines `bool` as `__vector __bool int`, causing ~30 type errors throughout the file.

## Reproducer

Hardware: IBM POWER8 S824 (ppc64le, ISA 2.07)
Compiler: GCC 10.5.0, Ubuntu 20.04

```bash
gcc-10 -maltivec -mvsx -std=c11 -DHACL_CAN_COMPILE_VEC128 -O2 -c Hacl_Hash_Blake2s_Simd128.c
```

Errors (~30 total):
```
Hacl_Hash_Blake2s_Simd128.c:58:7: error: used vector type where scalar is required
Hacl_Hash_Blake2s_Simd128.c:1228:64: error: incompatible types when initializing type '_Bool' using type '__vector __bool int'
```

## Root Cause

GCC with `-maltivec` adds `bool` as a keyword for `__vector __bool int`. When combined with `-std=c11`, this overrides `stdbool.h`'s `bool` mapping in struct access and function parameter contexts.

## Suggested Fix

Replace `bool` with `_Bool` throughout the SIMD128 files compiled with AltiVec. `_Bool` is the C99/C11 keyword that GCC does not override.

## Context

Discovered while enabling SIMD128 BLAKE2 in CPython on PowerPC (CPython PR #146118). The vec128 intrinsics in `libintvector.h` work correctly on POWER8 — only the higher-level BLAKE2 code has the bool issue.

POWER8/9 systems are in active production (IBM Cloud, enterprise, HPC). ppc64le is Tier 1 for Ubuntu, RHEL, SUSE, Debian.

--- a/lib/c/libintvector.h
+++ b/lib/c/libintvector.h
@@ -802,6 +802,18 @@ vector128 Lib_IntVector_Intrinsics_vec128_xor(vector128 x0, vector128 x1) {
 #if defined(HACL_CAN_COMPILE_VEC128)
 
 #include <altivec.h>
+
+/* GCC's AltiVec extension hijacks 'bool' as '__vector __bool int'.
+   Restore C99/C11 scalar bool for HACL* code. */
+#if defined(__GNUC__) && !defined(__clang__)
+#undef bool
+#define bool _Bool
+#undef true
+#define true 1
+#undef false
+#define false 0
+#endif
+
 #include <string.h> // for memcpy
 #include <stdint.h>
 

